debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/FindAllModulesCommand.java

```diff
@@ -0,0 +1,86 @@
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.vm29.tools.ddrinteractive.commands;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.ObjectModel;
+import com.ibm.j9ddr.vm29.j9.Pool;
+import com.ibm.j9ddr.vm29.j9.SlotIterator;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ModulePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9PoolPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+
+/**
+ * FindAllModules command displays all the modules loaded by the runtime
+ * 
+ * Example:
+ * 			!findallmodules
+ * Example output: 
+ * 			jdk.javadoc			!j9module 0x000001305F474498
+ * 			jdk.attach			!j9module 0x000001305F478798
+ * 			java.prefs			!j9module 0x000001305F478928
+ *			.....(A list of every module loaded by the runtime)
+ */
+
+public class FindAllModulesCommand extends Command 
+{
+	public FindAllModulesCommand() 
+	{
+		addCommand("findallmodules", "", "Outputs all the modules loaded by the runtime");
+	}
+
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException 
+	{
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			J9PoolPointer pool = vm.modularityPool();
+			Pool<J9ModulePointer> modulePool = Pool.fromJ9Pool(pool, J9ModulePointer.class);
+			SlotIterator<J9ModulePointer> poolIterator = modulePool.iterator();
+			J9ModulePointer modulePtr = null;
+			while (poolIterator.hasNext()) {
+				modulePtr = poolIterator.next();
+				/*
+				 * Since both packages and modules could be in the pool,
+				 * the iterator checks on the region and make sure that it is
+				 * pointing to a module. (For modules, the pointer must be in heap)
```

Comment:
try to be more specific and mention that we distinguish between module and j9package by checking if the first field is in the heap. Since the first field for j9module is an object and the first field for j9package is a J9UTF8